### ISSUE TITLE
Handle Liquisoap < 1.4.4 on bullseye and focal

### ISSUE THREAD

**Issue by jooola** [LibreTime/libretime#1338]
On our path to support bullseye and focal, liquidsoap is giving us a hard time.

Bullseye has liquidsoap 1.4.3
Focal has liquidsoap 1.4.1

I open this ticket to track all the changes we need to do in order to make liquidsoap work on bullseye and focal.

- https://github.com/savonet/liquidsoap/commit/68b13f2644dfaffc65718add31d8d87b9201b82e as we currently don't have a way to debug anything easily.
- Use `liquidsoap --version` to get the version instead of mystic script.
- Remove weird `airtime-liquidsoap` argument from liquidsoap execl arguments. https://github.com/LibreTime/libretime/blob/daca0f31664ad347e47f18e2d00aa5ad29b4f82f/python_apps/pypo/liquidsoap/__main__.py#L39
- Fix deprecation warnings: https://github.com/LibreTime/libretime/blob/daca0f31664ad347e47f18e2d00aa5ad29b4f82f/python_apps/pypo/liquidsoap/1.4/ls_script.liq#L16 should be `set("harbor.bind_addrs", ["0.0.0.0"])`
- More to come, that's for sure.

In addition, we could:
- patch most of the fixes from 1.4.1 to 1.4.4 into bullseye and focal. What would be awesome is if some magic would let us upload 1.4.4 into bullseye, don't know if Ubuntu has a stricter policy.
- patch the fixes ourself in our liquidsoap scripts, don't know if liquid soap is as flexible as python to overwrite var or function ?
- use an installation method than apt, maybe opam ?

**Comment by paddatrapper** [comment]
> patch most of the fixes from 1.4.1 to 1.4.4 into bullseye and focal. What would be awesome is if some magic would let us upload 1.4.4 into bullseye, don't know if Ubuntu has a stricter policy.

Same policy. Only critical bug fixes and security fixes get backported to stable releases. There is the -backports repo that we could potentially use, but need to see how many of the dependencies are stable/need to be backported. The other issue with backports is that it needs to track testing, so would get updated to newer versions as they get to testing and we loose the matching.

> patch the fixes ourself in our liquidsoap scripts, don't know if liquid soap is as flexible as python to overwrite var or function ?

We will have to do this. It doesn't really support it, which is why we have multiple versions of the scripts. See the #1138 for the changes required for 1.4.1. I don't see the strict requirement for https://github.com/savonet/liquidsoap/commit/68b13f2644dfaffc65718add31d8d87b9201b82e. It is a nice to have, sure, but it is not critical for 1.4.x support. The other points are things that can be fixed in our liquidsoap scripts.

 I do not want to introduce more packaging systems. We're trying to move away from pip installing python packages (thanks to things changing out from under us), I don't see the value in adding another one where liquidsoap can change out from under us. Targeting 2 or 3 liquidsoap versions where they are typically the same minor version is less error-prone. 1.4.1 and 1.4.3 should be API compatible - https://github.com/LibreTime/libretime/issues/1136#issuecomment-749019118

Even if we roll our own apt repo, there will always be a miss-match at some point when a distro ships a liquidsoap version higher than what we support and we need to update liquidsoap in the project before we can update the repo. Then users get updated liquidsoap without updating LT (for example unattended-upgrades, enabled by default on Debian and Ubuntu) and have their playout break.

**Comment by paddatrapper** [comment]
Ah, this is the fix I used - https://github.com/LibreTime/libretime/commit/68cb56be75a72bfd695cf5b7b699464eaf3dbecf

**Comment by paddatrapper** [comment]
The only bug here is the cause of this

> Use liquidsoap --version to get the version instead of mystic script.

So, changing this to a feature request

**Comment by jooola** [comment]
@paddatrapper I think I faced more problems than just this version issue, hence the idea of wrapping everything in a ticket.

But let me try a fresh install with the fix, and maybe I won't face the issues I had before.

**Comment by paddatrapper** [comment]
Ok, let me know and/or just update the labels if you find anything
